text/0000-slatepack.md

```diff
@@ -0,0 +1,391 @@
+  - `SlatePack Message` is an ascii-armor string that supports encryption of its payload with a `SlatePack Address`
+    - An encrypted `SlatePack Message` is not meaningfully larger than a plain text `SlatePack Message` with regard to transportability as proposed here
+
+4. _The difference between synchronous and asynchronous transaction methods is abstracted away from the end user with the SlatePack standard_
+  - `grin-wallet send -d SlatePackAddress 1.337` will first try to send the Grin synchronously via Tor to the `SlatePack Address`
+  - If that fails it will fall back to outputting an armored encrypted `SlatePack Message` string for manual copy and paste transport
+  - Example `SlatePack Address`: `slatepack1p4fuklglxqsgg602hu4c4jl4aunu5tynyf4lkg96ezh3jefzpy6swshp5x`
+
+5. _Asynchronous transactions are now encrypted by default by knowing the `SlatePack Address` of your counterparty(s)_
+  - If a counterparty is unwilling or unable to provide a `SlatePack Address`, a plain text `SlatePack Message` can still be exchanged
+
+6. _Sending a mobile Grin transaction should be as easy as scanning a simple QR encoded from a bech32 `SlatePack Address`_
+  - Or as easy as pasting the `SlatePack Address` of your counterparty into your wallet for any other device
+  - Or if Tor is not accessible, or the receiving party is not online, as easy as copying and pasting a couple of `SlatePack Message` strings with a counterparty in an alternative communication channel (email, forum, social media, instant messenger, generic web text box, carrier pigeon etc.)
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The SlatePack standard defines the three primary components: `SlatePack Address`, `SlatePack Message` and `SlatePack Workflow`.
+
+The `SlatePack Address` is a shareable bech32 encoded ed25519 public key that can be used both to _route synchronous transactions_ and to _encrypt asynchronous transactions_.
+
+The `SlatePack Message` is an easily copy and pastable ascii-armor string that contains an encrypted slate payload by default and is used in asynchronous transactions.
+
+The `SlatePack Workflow` specifies how both of these components interact in a universally adoptable transaction standard for Grin.
+
+## `SlatePack Address`
+
+A `SlatePack Address` is a bech32 encoded ed25519 public key and when shared with other parties is used to represent the ability to receive Grin transactions.
+
+grin-wallet already handles ed25519 keys for the v3 onion addresses in Tor transactions. These keys can be extended to be a general `SlatePack Address` to allow a universal key format for both transport and encryption that is error-checked, QR friendly and easily human identifiable.
+
+  - Existing ed25519 public keys from the wallet are bech32 encoded with `slatepack` as the `human-readable part` to build a `SlatePack Address`
```

Comment:
I think it is more natural for cryptocurrency users to pass around bech32 addresses instead of onion addresses (which may or may not have 'http://' and '.onion' framing). 

It should hopefully be easier for users and services to _think_ about how to handle this (as a comprehensive tx method as opposed to a manual Tor attempt and string fallback- while these are essentially the same I think one is easier for a user to digest). When you give another grin user this address, it isn't just a routing address it is an encryption key, which could be confusing as just a .onion. 

The other advantage is in places where Tor is less culturally acceptable, using bech32 addresses in place of onion addresses can potentially give a meaningfully different perception despite the similarities under the hood. It is also more efficient for QR encoding.

Reply:
To clarify, I meant specifically without the http:// & .onion

Reply:
To follow up with a few other advantages with bech32 over onionv3 encoding of an ed25519 key:
- we get a human readable prefix that can distinguish mainnet/floonet/potential future non-slatepack bech32 grin payment addresses
- it can help distinguish a standard hidden service address from an address meant to receive grin (of course most of this is done in context but you could imagine a hidden service that wants to share mirror URLs and slatepack addresses in the same signed message)
- bech32 is becoming a commonly used key format on other networks and platforms meaning a potentially reduced cognitive load for a user keeping track of an address to receive grin payments among others